Source/Core/VideoCommon/TextureCacheBase.cpp

```diff
@@ -1250,12 +1275,7 @@ void TextureCacheBase::CopyRenderTargetToTexture(u32 dstAddr, EFBCopyFormat dstF
   bool copy_to_ram = !g_ActiveConfig.bSkipEFBCopyToRam;
   bool copy_to_vram = true;
 
-  if (copy_to_ram)
-  {
-    EFBCopyParams format(srcFormat, dstFormat, is_depth_copy, isIntensity);
-    CopyEFB(dst, format, tex_w, bytes_per_row, num_blocks_y, dstStride, srcRect, scaleByHalf);
-  }
-  else
+  if (!copy_to_ram)
```

Comment:
The order was important here. Do you update the "hash" of the vram copy on the first CPU access? Else we might fail to use the upscaled version.

Reply:
Yep. The hash is set to 0 and the needs_rehash flag is set to false, which means no hash calculation will occur when the copy is used by the GPU. When the CPU touches it, or it's flushed, the hash is recalculated.

Reply:
Nice